awx/main/notifications/webhook_backend.py

```diff
@@ -61,23 +63,60 @@ def format_body(self, body):
 
     def send_messages(self, messages):
         sent_messages = 0
+        retries = 0
         if self.http_method.lower() not in ['put', 'post']:
             raise ValueError("HTTP method must be either 'POST' or 'PUT'.")
         chosen_method = getattr(requests, self.http_method.lower(), None)
+        resp = None
+        '''
+        Sometimes we hit redirect URLs. We must account for this. We still extract the redirect URL from the response headers and try again. Max retires == 5
+        '''
         for m in messages:
+
             auth = None
             if self.username or self.password:
                 auth = (self.username, self.password)
-            r = chosen_method(
-                "{}".format(m.recipients()[0]),
-                auth=auth,
-                data=json.dumps(m.body, ensure_ascii=False).encode('utf-8'),
-                headers=dict(list(get_awx_http_client_headers().items()) + list((self.headers or {}).items())),
-                verify=(not self.disable_ssl_verification),
-            )
-            if r.status_code >= 400:
-                logger.error(smart_str(_("Error sending notification webhook: {}").format(r.status_code)))
-                if not self.fail_silently:
-                    raise Exception(smart_str(_("Error sending notification webhook: {}").format(r.status_code)))
-            sent_messages += 1
+
+            # convert emtpy string to empty dict for POST/PUT
+            if not m.body:
+                m.body = dict()
+
+            url = "{}".format(m.recipients()[0])
```

Comment:
Python modernization: let's just do `str(m.recipients()[0])`.  Even the `str(...)` part is probably unnecessary, tbh.